### ISSUE TITLE
Window rotation, function names don't match the rotation direction

### ISSUE THREAD

**Issue by duianto** [syl20bnr/spacemacs#6868]
### Description

There are two spacemacs window rotation functions:

The first function:
`spacemacs/rotate-windows`
https://github.com/syl20bnr/spacemacs/blob/develop/layers/%2Bdistributions/spacemacs-base/funcs.el#L231

is bound to: SPC w r
and it rotates each window backwards,
based on the window numbers,
1 > 3 > 2 > 1 > ...

but it's comment states that:

> "... Giving a negative prefix
> argument takes the kindows rotate backwards. .."

The second function: 
`spacemacs/rotate-windows-backward`
https://github.com/syl20bnr/spacemacs/blob/develop/layers/%2Bdistributions/spacemacs-base/funcs.el#L260

is bound to: SPC w R
and it rotates in the opposite direction of the first,
currently it's forwards,
1 > 2 > 3 > 1 > ...

but it's name and comment:

>   "Rotate your windows backward."

states that it rotates backwards.

The `spacemacs/rotate-windows` function should probably be rewritten to rotate forwards.
### System info:

win 10 v1511, b10586.545
emacs 24.5.1
spacemacs develop branch (downloaded 20160811)
Distribution: spacemacs
Editing style: vim
Completion: helm
### Steps to reproduce
- Start Emacs + Spacemacs
- Split the frame into two windows: 
- SPC w S  (Split-window-below-and-focus)
- Change the buffer to something else, 
-  e.g., SPC b n  (next-buffer) (by default it changes to the _scratch_ buffer)
- Split again: SPC w S
- Change to another buffer than the once in the first two windows,
-  e.g., SPC b b mes RET (it opens the _Messages_ buffer)
- Rotate: SPC w r  (spacemacs/rotate-windows)
- take note of the rotation direction
- Rotate in the other direction SPC w R (spacemacs/rotate-windows-backward)
- it rotates in the opposite direction but it's not backwards.
### Observed Behavior

SPC w r  (spacemacs/rotate-windows)
rotates each window backwards to the previous window number.
and
SPC w R  (spacemacs/rotate-windows-backward)
rotates each window forwards to the next window number.
### Expected Behavior

SPC w r  (spacemacs/rotate-windows)
should rotate each window forwards to the next window number.
and
SPC w R  (spacemacs/rotate-windows-backward)
should rotate each window backwards to the previous window number.
#### Consistency

It seems like a lot of lowercase keybindings operate forwards / to the next occurrence,
and uppercase keybindings operate backwards / to the previous occurrence.
e.g., 
C-w C-r, C-w r
(evil-window-rotate-downwards)
this function rotates forwards
and
C-w C-S-r, C-w R
(evil-window-rotate-upwards)
this function rotates backwards

n  (evil-search-next)
N  (evil-search-previous)
f  (evil-find-char)
F  (evil-find-char-backward)
p  (evil-paste-after)
P  (evil-paste-before)
### Function names

If the `spacemacs/rotate-windows` function is rewritten to rotate forwards,
then maybe it should be renamed to specify it's rotation direction.

to something like this:
`spacemacs/rotate-windows-forwards`

and the function:
`spacemacs/rotate-windows-backward`

is missing an "s" at the end,
with the "s" added it becomes:
`spacemacs/rotate-windows-backwards`
#### Keybindings, function comments, documentation and transient state keys

When it's decided if the `spacemacs/rotate-windows` function 
is going to be rewritten, and both function names changed.

Then i'll make a PR with the updated,
keybindings, function comments, and documentation.

I'll also make separate issue for the
Window Manipulation Transient State popup window,
with a couple of different ways that the rotate forwards 
and backwards key descriptions could be written.

**Comment by bmag** [comment]
I agree with this suggestion. I never noticed that `spacemacs/rotate-windows` rotates the windows in the reverse order, but now that you pointed it out it's kind of obvious that it's wrong. It's especially notable when the frame is divided into 4 windows like this, and window 4 moves to 3 instead of 1.
```
+-----+-----+
|  1  |  2  |
+-----+-----+
|  3  |  4  |
+-----+-----+
```
I also agree that we should rename `spacemacs/rotate-windows` to match `spacemacs/rotate-windows-backward`. However we don't need the added `s` to the name, for example compare with `isearch-forward`. Also searching `SPC h d f forward$` has much more results than `forwards$`.

While we're at it, we can improve the implementation of `spacemacs/rotate-windows`: consider using `window-state-get` instead of `window-buffer` and `window-start`, and `window-state-put` instead of `set-window-buffer` and `set-window-start`. Window-states save more than just `window-start` and `window-buffer`, and I hope they won't change the size of the windows when rotating.

PR welcome.

P.S. about formatting: there are random line-breaks in the text. You don't need to press \<return\> in the middle of sentences to format your text, it makes it harder to read in other screen resolutions (I think that's what you did).

P.S. 2: you can use `SPC h I` to open a new issue report with some info auto-generated, e.g. the system info.

**Comment by duianto** [comment]
I didn't know that it looked strange in other screen resolutions, that's good to know, i'll avoid adding manual line-breaks in the future.

When i pressed `SPC h I` then the messages buffer showed:
`apply: Searching for program: no such file or directory, git`

I downloaded portablegit-2.9.3-64-bit, unpacked it and added the path to the PortableGit-2.9.3-64-bit\cmd folder, to the windows system environment PATH variable, and now when i press `SPC h I `
then a REPORT_SPACEMACS_ISSUE buffer opens, but in the System info section:

#### System Info :computer:
- OS: windows-nt
- Emacs: 24.5.1
- Spacemacs: 0.105.21
- Spacemacs branch: nil (rev. nil)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm emacs-lisp)
```

the spacemacs branch and rev. "id", shows: nil

Maybe i have to download a local version of the spacemacs repository, using the windows cmd, git commands or with the git-gui app, that might be why emacs doesn't see a branch, rev "id".

**Comment by duianto** [comment]
I solved the issue that caused the system info, spacemacs branch to show `nil`.

I cloned the spacemacs repository to the folder:
c:\users\username\appdata\roaming\

With the command:
git clone https://github.com/syl20bnr/spacemacs .emacs.d

This command might also work:
git clone https://github.com/syl20bnr/spacemacs %homepath%/Appdata/Roaming/.emacs.d
then one doesn't have to navigate to the roaming folder first.

After that `SPC h d s` `(spacemacs/describe-system-info)` outputs:
- Spacemacs branch: master (rev. 9f9faa4)

That works, but i wanted to use the develop branch, so after a bunch of googling and testing, i ended up with this command:
git checkout -b develop origin/develop

And now `SPC h d s` outputs: 
- Spacemacs branch: develop (rev. d94d0d9)

I know it isn't related to the rotate windows issue, but someone else might run into the same `nil` problem, and stumble over this comment.


Now regarding the rewrite/renaming of the rotate windows function, i hope i didn't give the impression that i was going to rewriting the rotate function. Only that i could write PRs for the comment, documentation, and keybinding, but the keybinding should probably be changed at the same time as when the function is renamed.

I don't have the lisp knowledge to rewrite the function, i have only used emacs for three months and spacemacs for about a month.
